src/test/java/org/jabref/logic/bibtex/comparator/GroupDiffTest.java

```diff
@@ -0,0 +1,62 @@
+package org.jabref.logic.bibtex.comparator;
+
+import java.util.Optional;
+
+import org.jabref.model.groups.AllEntriesGroup;
+import org.jabref.model.groups.ExplicitGroup;
+import org.jabref.model.groups.GroupHierarchyType;
+import org.jabref.model.groups.GroupTreeNode;
+import org.jabref.model.metadata.MetaData;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class GroupDiffTest {
+
+    private final MetaData originalMetaData = mock(MetaData.class);
+    private final MetaData newMetaData = mock(MetaData.class);
+    private GroupTreeNode rootOriginal;
+
+    @BeforeEach
+    void setup() {
+        rootOriginal = GroupTreeNode.fromGroup(new AllEntriesGroup("All entries"));
+        rootOriginal.addSubgroup(new ExplicitGroup("ExplicitA", GroupHierarchyType.INCLUDING, ','));
+        GroupTreeNode parent = rootOriginal
+                .addSubgroup(new ExplicitGroup("ExplicitParent", GroupHierarchyType.INDEPENDENT, ','));
+        parent.addSubgroup(new ExplicitGroup("ExplicitNode", GroupHierarchyType.REFINING, ','));
+    }
+
+    @Test
+    void compareEmptyGroups() {
+        when(originalMetaData.getGroups()).thenReturn(Optional.empty());
+        when(newMetaData.getGroups()).thenReturn(Optional.empty());
+
+        assertEquals(Optional.empty(), GroupDiff.compare(originalMetaData, newMetaData));
+    }
+
+    @Test
+    void compareGroupWithItself() {
+        when(originalMetaData.getGroups()).thenReturn(Optional.of(rootOriginal));
+        when(newMetaData.getGroups()).thenReturn(Optional.of(rootOriginal));
+
+        assertEquals(Optional.empty(), GroupDiff.compare(originalMetaData, newMetaData));
+    }
+
+    @Test
+    void compareWithChangedGroup() {
+        GroupTreeNode rootModified = GroupTreeNode.fromGroup(new AllEntriesGroup("All entries"));
+        rootModified.addSubgroup(new ExplicitGroup("ExplicitA", GroupHierarchyType.INCLUDING, ','));
+
+        when(originalMetaData.getGroups()).thenReturn(Optional.of(rootOriginal));
+        when(newMetaData.getGroups()).thenReturn(Optional.of(rootModified));
+
+        Optional<GroupDiff> groupDiff = GroupDiff.compare(originalMetaData, newMetaData);
```

Comment:
Would be nice to compare the conents here as well, e.g. createa GroupDiff object as expected

Reply:
Hi @Siedlerchr 
thanks for reviewing!
That's what I first wanted to do, but it would require to change GroupDiff's constructor to `protected`. Right now, the constructor is `private` and only invoked when the `compare` method is executed. Would that be fine if I change it to `protected`?

Reply:
I think, it's package private (without any prefix like private, public, ...)